Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/tabs/location/ContactDiaryLocationListViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package de.rki.coronawarnapp.contactdiary.ui.day.tabs.location
+
+import androidx.lifecycle.asLiveData
+import com.squareup.inject.assisted.Assisted
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.contactdiary.model.Location
+import de.rki.coronawarnapp.contactdiary.storage.ContactDiaryRepository
+import de.rki.coronawarnapp.contactdiary.util.SelectableItem
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import org.joda.time.Instant
+
+class ContactDiaryLocationListViewModel @AssistedInject constructor(
+    @Assisted selectedDay: Long,
+    contactDiaryRepository: ContactDiaryRepository
+) : CWAViewModel() {
```

Comment:
Inject the `dispatcherProvider` otherwise the `launch` will run on the UI thread.